test/parser.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { parser } from '../src/parser';
+
+const outputJSONMessagesChannels =
```

Comment:
I will remove the `outputJSONMessagesChannels` and use snapshots instead. Is that alright?

Reply:
I wouldn't recommend using snapshots.

Reply:
oh ok. Will not use snapshots now. What's the reason for not using snapshots?

Also, I'm not sure about what "fixtures" file is. Is it the same thing as moving the outputs to a different file and then importing them ?

Reply:
> oh ok. Will not use snapshots now. What's the reason for not using snapshots?

Snapshots test the function's result with latest saved result from this function, so if you will change something from logic and then you will run the tests you probably will have broken tests, so... you'd better use a small spec and test it with a parsed string as you are currently doing or test a slice of the parsed spec :) You can read about problems with snapshots here https://medium.com/@tomgold_48918/why-i-stopped-using-snapshot-testing-with-jest-3279fe41ffb2

> Also, I'm not sure about what "fixtures" file is. Is it the same thing as moving the outputs to a different file and then importing them ?

The purpose of a test fixture is to ensure that there is a well known and fixed environment in which tests are run so that results are repeatable. Some people call this the test context, so usually this is a prepared data to test.

Reply:
Thanks for the explanations :)

Reply:
@vinitshahdeo Please check if I have the "fixtures" file done correctly. I was a bit unsure about that.